File: Divisores.py
import psycopg2
from tabulate import tabulate
def  post(x,y):
    try:
        conexion = psycopg2.connect(
            host = "localhost",
            port = "5432",
            user = "postgres",
            password = "jblue5queso",
            dbname = "tarea1"
            )
    except:
        print("Sin Conexion Exitosa\n")
            
    cursor = conexion.cursor()
    cursor.execute("INSERT INTO divisores(numero_in, divisores) VALUES (%s, %s);", (x, y))
    conexion.commit()
    cursor.close()
    conexion.close()

mat=[]
print("Introduce un número para mostrar sus divisores: ")
x=int(input())
for i in range(1, x+1):
    if x%i==0:
     mat.append(i)
print("Los divisores del numero son: \n ",mat)
post(x,mat)
    

File: Divisores.m
pkg load database;
clc; 
clear all; 
close all;

fprintf('Bienvenido ');
n=input('Introduce un numero para mostrar sus divisores: ');
while (n~=fix(n))||(n<0)
  fprintf('Ingrese un valor numerico ');
  n=input(',Digite otro valor de n: ');
end
cont=1;
for i=1:n
    x=mod(n,i);
    if x==0
        V(cont)=i;
        cont=cont+1;    
    endif
end    
fprintf('Los divisores de %i son: ',n)
disp(V)

%Conectar con PGADMIN
linearecta=mat2str(V);
params= cell(1,2);
params{1,1}=n;
params{1,2}=linearecta;


conn = pq_connect(setdbopts('dbname','tarea1','host','localhost',
'port','5432','user','postgres','password','jblue5queso'));
N=pq_exec_params(conn, "insert into divisores(numero_in,divisores) values($1,$2);",params); %insertar datos en la tabla

%Texto
save('Divisores.txt','V','-ascii');